akka-remote/src/main/scala/akka/remote/artery/compress/TopHeavyHitters.scala

```diff
@@ -205,6 +219,7 @@ private[remote] final class TopHeavyHitters[T >: Null](val max: Int)(implicit cl
   private def insertKnownNewHeavy(hashCode: HashCodeVal, item: T, count: Long): Unit = {
     removeHash(lowestHitterIndex)
     lowestHitterIndex = insert(hashCode, item, count)
+    takeSnapshot()
```

Comment:
ok, so the snapshot is only taken when there is a new entry added to the table, and that will normally be stable unless lots of temporary (short lived) actors are used. I just want to be sure that we don't take the snapshot on every hit update.

Later, when we move the ownership to the Decoder stage the periodic announcement task is scheduled by the stage and then it can also take the snapshot at that point. Some counting if snapshot is really needed might be a good idea anyway if snapshotting is costly. I'm pretty sure we already compare snapshots before sending the announcements to avoid sending identical.